enhancements/hypershift/performanceprofilecontroller-node-tuning.md

```diff
@@ -361,6 +398,11 @@ See openshift/hypershift#1782
 To be able to handle `KubeletConfig` propagation properly NodePool controller should be changed to recognize a `KubeletConfig` as a valid content and set its defaults properly, that is mainly to ensure label `machineconfiguration.openshift.io/role=worker` is set.
 This can be made by changing [`defaultAndValidateConfigManifest`](https://github.com/openshift/hypershift/blob/fa0ca3d09fab02ebff64d45b97cc1abaf4f1c27a/hypershift-operator/controllers/nodepool/nodepool_controller.go#L1439)  to handle `KubeletConfig` almost as it is handling `MachineConfig` right now.
 
+To view if the performance profile has been applied successfully, NodePool controller will watch the performance profile status configmap located in the hosted control plane namespace, and updates the PerformanceProfileAppliedSuccessfully condition under NodePool.status.conditions.
```

Comment:
Will the nodepool only update one condition from the performance profile status? Will there ever be more than one condition?

1) If only one condition: Maybe the configmap data can just include keys for each of the condition fields (type, msg, status, reason, etc) instead of requiring an encode/decode.

2) If more than one condition: It might make sense to simply namespace the condition type with a prefix like `PerformanceProfile/` (ie. `PerformanceProfile/PerformanceAppliedSuccessfully`) and just copy all the conditions from the serialized status.

Reply:
@csrwng On second thought, I think we would like to proceed with your first suggestion (removing the encode/decode).... looking forward to your approval!

Reply:
@csrwng What about adding a sub filed under the node pool status
`PerformanceProfileStatus` that will hold all the information about the Performance Profile status?
Then we could simply access it as follow: 
`nodepool.status.performanceProfileStatus` 
I saw there is something similar for the platform status: https://github.com/openshift/hypershift/blob/485a176b8bed72f44f2fe7958d7468441a5cb02f/vendor/github.com/openshift/hypershift/api/hypershift/v1beta1/nodepool_types.go#L202

Reply:
@Tal-or hey catching up. I was out on PTO.
We have the pattern in HCP conditions of a prefix (as in the cluster network operator conditions):
```
    - lastTransitionTime: "2024-05-30T20:55:45Z"
      message: ""
      reason: AsExpected
      status: "False"
      type: network.operator.openshift.io/Degraded
    - lastTransitionTime: "2024-05-30T20:55:02Z"
      message: ""
      reason: AsExpected
      status: "True"
      type: network.operator.openshift.io/Upgradeable
    - lastTransitionTime: "2024-05-30T20:55:04Z"
      message: All is well
      observedGeneration: 1
      reason: AsExpected
      status: "True"
      type: IgnitionServerValidReleaseInfo
    - lastTransitionTime: "2024-06-14T20:45:04Z"
      message: ""
      reason: AsExpected
      status: "False"
      type: network.operator.openshift.io/Progressing
    - lastTransitionTime: "2024-05-30T20:55:08Z"
      message: ""
      reason: AsExpected
      status: "True"
      type: network.operator.openshift.io/Available
```
My preference would be to do that instead of adding a separate field.

Reply:
Agreed, we'll go with this approach

Reply:
@rbaturov FYI ^